presto-spark-base/src/main/java/com/facebook/presto/spark/execution/PrestoSparkOutputOperator.java

```diff
@@ -285,9 +313,38 @@ public Page getOutput()
     @Override
     public void finish()
     {
+        flush();
+        updateMemoryContext();
         finished = true;
     }
 
+    private void flush()
+    {
+        if (currentBatchSize > 0) {
+            verify(currentBatch != null);
+            // Uses currentBatch internally. Must be called before currentBatch is set to null.
+            int rowsListRetainedSize = getCurrentBatchRetainedBytes();
+            List<PrestoSparkRow> rowsList = currentBatch.build();
+            BufferedRows bufferedRows = new BufferedRows(rowsList, rowsListRetainedSize);
+            rowBuffer.enqueue(bufferedRows);
+            currentBatch = null;
```

Comment:
nit: Any reason why defer the creation of `ImmutableList.Builder` to `appendRow`? Because I would personally eagerly create the new `ImmutableList.Builder` here.

Reply:
Just in case no more rows will be appended, it saves an allocation. No big difference though.